### PR TITLE
openafs: Patch for Linux kernel 5.18

### PR DESCRIPTION
**Pull request by andersk**
###### Description of changes

Cherry-pick the same patches used by the Debian package for Linux kernel 5.18 support.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by spacefrogg — APPROVED** [review]
Thanks! The changes are looking good!

Also thanks for also taking up the responsibility of maintaining and Welcome! :)

File: pkgs/servers/openafs/1.8/module.nix [hunk 1]
```diff
@@ -38,21 +38,21 @@ stdenv.mkDerivation {
...
-    # Add autoconf-archive to src/external
-    (fetchBase64Patch {
-      url = "https://gerrit.openafs.org/changes/14942/revisions/006616bd8e88b2d386a5ddc23973cf3e625cb80d/patch";
-      hash = "sha256-55sc2sKy5XkQHAv6ysVxi69+0xVsHnN2TS144UTeLHU=";
-    })
     # Import of code from autoconf-archive
-    (fetchBase64Patch {
-      url = "https://gerrit.openafs.org/changes/14943/revisions/d3782b1d4e6fd81c5432e95112eb44305f07f272/patch";
+    (fetchpatch {
+      url = "https://git.openafs.org/?p=openafs.git;a=patch;h=d8205bbb482554812fbe66afa3c337d991a247b6";
```
Comment (SuperSandro2000) [inline pkgs/servers/openafs/1.8/module.nix:43]: You are sure this hash still matches?

**Review by SuperSandro2000 — COMMENTED** [review]
Why do we update 1.8 only? What is with 1.9?

File: pkgs/servers/openafs/1.8/module.nix (hunk 1 above)
Comment (andersk) [reply pkgs/servers/openafs/1.8/module.nix:43]: Yes.

**Comment by andersk** [comment]
> Why do we update 1.8 only? What is with 1.9?

1.9 is a development branch. According to [upstream](https://www.openafs.org/): “OpenAFS 1.9.0 is the first in a series of OpenAFS development releases intended to facilitate testing of new features. It should not be considered production-ready.”

There have been no upstream releases from that branch since 1.9.0. (A 1.9.1 release was cut but never announced.) It’s over a year older than 1.8.8.1. It’s much farther behind on Linux kernel support, and also missing a critical fix that’s needed to allow operation after UNIX timestamp 0x60000000 (2021-01-14 08:25:36 UTC). Porting patches to this branch would be a major effort, and that effort would be wasted since we can guarantee nobody is using it.

I’m honestly not sure why we have it in tree. If @spacefrogg has a use case for it, more power to them I guess, but I’m not going to divert my effort into maintaining a development branch when tracking down the right patches for stable is tricky enough.

**Comment by SuperSandro2000** [comment]
That sounds like we should drop 1.9 to simplify things if it isn't ready to use at all.

**Comment by spacefrogg** [comment]
Sure, we should drop 1.9, then. From the develop of 1.8, I was expecting it to go forward mainly on 1.9. This does not seem to happen.

**Comment by andersk** [comment]
Okay, I’ve opened #180154 dropping 1.9. (No need to block this 1.8 fix with that.)

**Pull request merged**